libs/src/services/creatorNode/index.js

```diff
@@ -22,18 +22,16 @@ class CreatorNode {
    * @param {string} endpoints creator node endpoints
    * @param {number} trackId
    */
-  static async checkIfDownloadAvailable (trackId, endpoints) {
-    for await (const endpoint of endpoints.split(',')) {
-      try {
-        const res = await axios({
-          baseURL: endpoint,
-          url: `tracks/download_status/${trackId}`
-        })
-        if (res.cid) return res.cid
-      } catch (e) {
-        console.log(`Unable to download ${trackId} from ${endpoint}`)
-        // continue
+  static async checkIfDownloadAvailable (endpoints, trackId) {
+    const primary = CreatorNode.getPrimary(endpoints)
```

Comment:
nit but question - isn't it better to use `this.fn()` when accessing fn in same class. not sure why you do `CreatorNode.fn()`

Reply:
this doesn't work. there's no instance since it's static

Reply:
of course, my b